### ISSUE TITLE
Non-blocking API calls

### ISSUE THREAD

**Issue by rakanalh** [raiden-network/raiden#1939]
## Problem Definition

Currently, the API calls for joining a network, creating/closing a channel take quite a while. Those calls also suffer a problem that they need to be uninterrupted or otherwise, the logic will not execute to completion. 
An example would be when creating a channel, the API call would have to:
1. Create the channel and wait for the `ChannelOpened`
2. Trigger the deposit logic and wait for the deposit event.

Which means that if we interrupt the call in between, the deposit will not happen.

Scenarios where this can be problematic / confusing:
1. When the user invokes a time-consuming functionality and then refreshes the web UI.
2. Same as above but using the APIs directly, the default client timeout would trigger and stop the request. So the user has to explicitly set the timeout.
3. When developing a mobile client where the user has to keep the app in the foreground/active until the channel / token network operation is finished.

## Solution
Non blocking tasks that could be triggered with a `wait=false` argument or something similar. This would enable the user to call the API and quickly get a response `201 ACCEPTED` and the Raiden node would continue to operate and process the request in the background.

**Comment by hackaugusto** [comment]
I assume the problem is that an HTTP request can be interrupted **and** because of that the greenlet which is processing the request is killed.

If that is the case, another possible solution is to **not** expose APIs which are not "atomic". channel open + deposit don't have a corresponding state change, so the problem describe in the issue can happen, but  if we only have APIs for open and another for deposit, this should not happen even if the Raiden node crashes.

**Comment by rakanalh** [comment]
Closing this issue.

Initially, the thought was to have the API endpoints return a result "token" which can be used to poll. 
Knowing that all requests spawn a new greenlet, the idea was that if the request gets interrupted (timeout from the client side as an example) would stop the greenlet which turned out to be a false statement.

There are 2 cases:
1. Requests that perform an action such as "open channel", "close channel" will work in the request's greenlet. The greenlet is **not** killed when the client closes the request socket so the greenlet will keep running even after the client lost interest in waiting for the result. The problem to really solve here is to special to open channel because:
a. The open channel transaction has to be sent / mined.
b. The token contract should be called to approve the deposit funds
c. The token network proxy should be called to set the total deposit.

This operation should be atomic and we have issues to resolve this problem (https://github.com/raiden-network/raiden/issues/4768, https://github.com/raiden-network/raiden/issues/4767, https://github.com/raiden-network/raiden/issues/1621)

2. The other case which is enabling sending concurrent transfers. This is currently already the case because a single transfer would spawn a greenlet that would context-switch with other greenlets due to a single transfer being multiple messages.

After having discussed this with Augusto we realized that:
1. Adding response token will result in more poll requests to Raiden. There's no real benefit here provided that Raiden provides the concurrency level to handle multiple requests at the same time.
2. As stated before, a request is bound to a greenlet but not vice versa. So when a request's socket is disconnected, the greenlet will stay alive and keep working.
